src/server/rpc/procedures/waterwatch/waterwatch.js

```diff
@@ -0,0 +1,199 @@
+let msgs = [];
+
+function sendNextMsg(socket){
+    // QUESTION is there a need to check for the socket ID? no need.
+    if (msgs.length < 1){
+        return;
+    }
+    socket.send(msgs.shift());
+    setTimeout(sendNextMsg,DELAY,socket);
+}
+
+//checks msg ownership, needs access to socket.
+function isMsgOwner(msg){
+    return msg.dstId = socket.roleId
+}
+
+function stopSendingMsgs($this){
+    socket = $this.socket;
+    response = $this.response;
+    if (msgs) {
+        // remove those with a different roleId | dont remove other's messages
+        msgs = msgs.filter(msg => {
+            return msg.dstId != socket.roleId;
+        });
+    }
+    trace(`Stopped sending messages to user with roleId ${socket.roleId}`);
+    response.send('Stopped sending messages.');
+}
+
+// notes: dont forget to return null in your stop since we are handling the response in here.
+//**********************************
+
+// factor our the bulk of the work to provide easily readable settings for users.
+// TODO rename the function
+function doTheWork(options,socket,response,msgType){
+
+    let url = baseUrl+encodeQueryData(options);
+    trace('requesting this url for data', url);
+    rp(url)
+        .then(data => {
+            // santize and send messages to user
+            trace("Received data back");
+
+            try {
+                data = JSON.parse(data);
+            } catch(e) {
+                trace('Non-JSON data...');
+                return response.send('Bad API Result: ' + data);
+            }
+
+            //clean and store the sections we want in a form of a message
+            data.value.timeSeries.forEach(item => {
+                theData = {
+                    siteName: item.sourceInfo.siteName,
+                    lat: item.sourceInfo.geoLocation.geogLocation.latitude,
+                    lon: item.sourceInfo.geoLocation.geogLocation.longitude,
+                    // varName: item.variable.variableName,
+                    // varDescription: item.variable.variableDescription,
+                    unit: item.variable.unit.unitCode,
+                    value: item.values[0].value[0].value
+                };
+                trace(theData);
+                msgs.push({
+                    type: 'message',
+                    msgType: msgType,
+                    dstId: socket.roleId,
+                    content: theData
+                });
+            });
+            trace('loaded ', data.value.timeSeries.length,'  items');
+            // filterout messages ( imp if having multiple people use the shared msgs var)
+            let myMsgs = msgs.filter(msg => {
+                return msg.dstId == socket.roleId && msg.msgType == msgType
+            })
+            response.send(`Sending ${myMsgs.length} messages of ${msgType}`);
+            // start sending messages - will send other user's messages too
+            // QUESTION: can you send to whatever socket.roleId you want from any source? yes can do
```

Comment:
I want it to be on record that this is going to start sending other user's messages too.